### DISCUSSION TITLE
🚀 Add Ability to send HTTP code with send_ajax_response

### DISCUSSION THREAD

**Discussion by dougblackjr** [ExpressionEngine/ExpressionEngine#4108] [Feature Requests]
### Suggest an idea for ExpressionEngine

**Is your feature request related to a problem? Please describe.**
As I'm using EE to build some API functionality related to other 3rd party apps, I am finding I need to add certain HTTP codes (particularly 201, 400, 403) for ajax responses. Currently, the `send_ajax_response` function in the Output library only allows for the data and whether to send as an error (500) or success (200).

**Describe the solution you'd like**
Would like to see the ability to set the HTTP code in that function. 

**Describe alternatives you've considered**
Right now, I'm accomplishing the same with setting the header and echoing the response. Far from ideal.

**Teachability, Documentation, Adoption, Migration Strategy**

I envision changing this function to look like this:

```php
public function send_ajax_response($msg, $error = false, $statusCode = 200)
{
    $this->enable_profiler(false);

    if ($error === true) {
        $statusCode = 500;
    }

    $this->set_status_header($statusCode);

    if (ee()->config->item('send_headers') == 'y') {
        header("Content-Type: application/json; charset=" . ee()->config->item('charset'));
    }

    exit(json_encode($msg));
}
```

I'm still not sure this is the _best_ route, so I'd love input. I am a bit concerned about backwards compatibility.

- [X] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by intoeetive** [comment]
@dougblackjr excellent idea!

I think it's not necessary to add third parameter, we can add a check whether 2nd parameter is a boolean or a 3-digit number

**Comment by dougblackjr** [comment]
@intoeetive Nice! I'll get to work on a PR

**Comment by dougblackjr** [comment]
@intoeetive Added a PR here: https://github.com/ExpressionEngine/ExpressionEngine/pull/4110

I can't add labels or anything, let me know if I need to do anything else on this.
